plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -79,4 +88,54 @@ public void testShowCreateTable()
                         format("   location = '.*/" + schemaName + "/region'\n") +
                         "\\)");
     }
+
+    @Test
+    public void testListInformationSchemaMetadataNotFound()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_metadata_not_found (col integer)");
+
+        HdfsEnvironment.HdfsContext context = new HdfsEnvironment.HdfsContext(getSession().toConnectorSession());
+        org.apache.hadoop.fs.Path metadataDir = new org.apache.hadoop.fs.Path(getMetadataDirectory("test_metadata_not_found"));
+        FileSystem fileSystem = HDFS_ENVIRONMENT.getFileSystem(context, metadataDir);
+        org.apache.hadoop.fs.Path metadataFile = getMetadataJsonFile(fileSystem, metadataDir);
+        org.apache.hadoop.fs.Path renamedMetadataFile = new org.apache.hadoop.fs.Path(metadataFile + ".renamed");
+
+        fileSystem.rename(metadataFile, renamedMetadataFile);
+
+        assertQueryFails("SELECT * FROM test_metadata_not_found", "Failed to (read file|open input stream for file): .*");
```

Comment:
it would be worth catching the exception and inspecting whether the cause of it is a `TrinoException` with error code `ICEBERG_METADATA_NOT_FOUND`

Reply:
assertQueryFails ensures the exception is TrinoException. I don't think code verification is helpful here. This line is just for making sure if the metadata is invalid or not.